src/controller/java/src/chip/tlv/TlvWriter.kt

```diff
@@ -0,0 +1,361 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    Copyright (c) 2019-2023 Google LLC.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+ 
+package chip.tlv
+
+import com.google.protobuf.ByteString
+import java.io.ByteArrayOutputStream
+
+/**
+ * Implements Weave/Matter TLV writer that supports all values and tags as defined in the Spec.
+ *
+ * See go/weave-tlv-format and go/chip-specs#tlv for specification.
```

Comment:
Update links here?
Also should we just keep `Matter` without referencing Weave?